### PR TITLE
Improve Result#inspect output

### LINKED ISSUE #79: Inconsistent formating in Result#inspect
```Ruby
#<Twingly::Search::Result:0x3fc870543c10 @posts, @number_of_matches_returned=1000, @number_of_matches_total=2847@incomplete_result=false>
```

https://github.com/twingly/twingly-search-api-ruby/blob/785d8604c2422eee7a1abc26422649fcdc57f861/lib/twingly/search/result.rb#L38-L43

and:

https://github.com/twingly/twingly-search-api-ruby/blob/785d8604c2422eee7a1abc26422649fcdc57f861/lib/twingly/livefeed/result.rb#L26-L35
We should probably do something like this instead:

```Ruby
[
  "@posts",
  "@number_of_matches_returned=#{self.number_of_matches_returned}",
  "@number_of_matches_total=#{self.number_of_matches_total}",
  "@incomplete_result=#{self.incomplete?}",
].join(", ")
```

### PR DESCRIPTION
**Pull request by jage**
* Consistent comma separation (#79)
* Simple specs, assumes all available instance methods should be printed
* Remove @-prefx from output
* Update examples

Close #79

### CODE REVIEW

**Comment by jage** [comment]
Also close #66 hehe (#79 is a duplicate issue)

File: spec/livefeed/result_spec.rb [hunk 1]
```diff
@@ -2,17 +2,31 @@
...
 
     context "before query has populated responses" do
       its(:posts) { should be_empty }
     end
+
+    describe "#inspect" do
+      subject { described_class.new.inspect }
+
+      instance_methods.each do |method|
+        it { should include(method.to_s) }
```
Comment (roback) [inline spec/livefeed/result_spec.rb]: I think we need something before and after the method name here. `posts` matches `number_of_posts` for example. Something like the following (disclaimer: I have no idea whether this works):

```suggestion
        it { should match(/ #{method}[,=]/) }
```

Comment (jage) [reply spec/livefeed/result_spec.rb]: Yeah, good catch! I was a bit hesitant to repeat the exact syntax here, but that looks reasonable.

File: spec/result_spec.rb [hunk 2]
```diff
@@ -5,15 +5,29 @@
...
   context "before query has populated responses" do
     its(:posts) { should be_empty }
     its(:all_results_returned?) { should be_truthy }
   end
+
+  describe "#inspect" do
+    subject { described_class.new.inspect }
+
+    instance_methods.each do |method|
+      it { should include(method.to_s) }
```
Comment (roback) [inline spec/result_spec.rb]: Same as https://github.com/twingly/twingly-search-api-ruby/pull/83/files#r269577064 (it's not a problem right now though as all instance methods here have unique names)

File: lib/twingly/search/result.rb [hunk 3]
```diff
@@ -35,12 +35,17 @@ def all_results_returned?
...
       def inspect
-        matches = "@posts, "
-        matches << "@number_of_matches_returned=#{self.number_of_matches_returned}, "
-        matches << "@number_of_matches_total=#{self.number_of_matches_total}"
-        matches << "@incomplete_result=#{self.incomplete?}"
+        instance_methods = [
+          "posts",
+          "number_of_matches_returned=#{self.number_of_matches_returned}",
+          "number_of_matches_total=#{self.number_of_matches_total}",
+          "incomplete_result=#{self.incomplete?}",
```
Comment (roback) [inline lib/twingly/search/result.rb]: I think we should remove this one as we have `incomplete?` below

```suggestion
```

File: spec/result_spec.rb (hunk 2 above)
Comment (jage) [reply spec/result_spec.rb]: 1c7200e

File: spec/livefeed/result_spec.rb (hunk 1 above)
Comment (jage) [reply spec/livefeed/result_spec.rb]: 1c7200e

**Pull request merged**